components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -452,93 +452,135 @@ private String getOIDCAuthzEndpoint(Map<String, String> authenticatorProperties)
     protected void processAuthenticationResponse(HttpServletRequest request, HttpServletResponse response,
                                                  AuthenticationContext context) throws AuthenticationFailedException {
 
-        try {
+        OAuthClientResponse oAuthResponse = generateOauthResponse(request, context);
 
-            OAuthAuthzResponse authzResponse = OAuthAuthzResponse.oauthCodeAuthzResponse(request);
-            OAuthClientRequest accessTokenRequest = getAccessTokenRequest(context, authzResponse);
+        // TODO : return access token and id token to framework
+        mapAccessToken(request, context, oAuthResponse);
+        String idToken = mapIdToken(context, request, oAuthResponse);
 
-            // Create OAuth client that uses custom http client under the hood
-            OAuthClient oAuthClient = new OAuthClient(new URLConnectionClient());
-            OAuthClientResponse oAuthResponse = getOauthResponse(oAuthClient, accessTokenRequest);
+        Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+        if (StringUtils.isBlank(idToken) && requiredIDToken(authenticatorProperties)) {
+            throw new AuthenticationFailedException(ErrorMessages.ID_TOKEN_MISSED_IN_OIDC_RESPONSE.getCode(),
+                    String.format(ErrorMessages.ID_TOKEN_MISSED_IN_OIDC_RESPONSE.getMessage(),
+                            getTokenEndpoint(authenticatorProperties),
+                            authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_ID)));
+        }
+
+        OIDCStateInfo stateInfoOIDC = new OIDCStateInfo();
+        stateInfoOIDC.setIdTokenHint(idToken);
+        context.setStateInfo(stateInfoOIDC);
 
-            // TODO : return access token and id token to framework
-            String accessToken = oAuthResponse.getParam(OIDCAuthenticatorConstants.ACCESS_TOKEN);
+        AuthenticatedUser authenticatedUser;
+        Map<ClaimMapping, String> claims = new HashMap<>();
+        Map<String, Object> jsonObject = new HashMap<>();
 
-            if (StringUtils.isBlank(accessToken)) {
-                throw new AuthenticationFailedException(ErrorMessages.ACCESS_TOKEN_EMPTY_OR_NULL.getCode(),
-                        ErrorMessages.ACCESS_TOKEN_EMPTY_OR_NULL.getMessage());
+        if (StringUtils.isNotBlank(idToken)) {
+            jsonObject = getIdTokenClaims(context, idToken);
+            if (jsonObject == null) {
```

Comment:
Is it possible to use the isEmpty for Map?
https://www.tabnine.com/code/java/methods/java.util.Map/isEmpty

Reply:
Fixed 76bde03e8a2edc30459231f28a0cf90d9bbee240